subworkflows/nf-core/bam_variant_calling_freebayes/main.nf

```diff
@@ -0,0 +1,43 @@
+include { SAMTOOLS_FAIDX } from '../../../modules/nf-core/samtools/faidx/main'
+include { FREEBAYES      } from '../../../modules/nf-core/freebayes/main'
+include { BCFTOOLS_INDEX } from '../../../modules/nf-core/bcftools/index/main'
+include { BCFTOOLS_SORT  } from '../../../modules/nf-core/bcftools/sort/main'
+
+workflow BAM_VARIANT_CALLING_FREEBAYES {
+
+    take:
+    ch_input        // channel: [ val(meta), bam/cram, bai/crai, bam/cram, bai/crai, bed ]
+    genome          // channel: /path/to/fasta
+    samples         // channel: /path/to/samples
+    populations     // channel: /path/to/populations
+    cnv             // channel: /path/to/cnv
+
+    main:
+
+    ch_versions = Channel.empty()
+
+    // Generate index for genome fasta
+    ch_genome = Channel.fromPath ( genome ).map { file -> [ [ id: file.baseName ], file ] }
+    SAMTOOLS_FAIDX ( ch_genome )
```

Comment:
Shouldn't we take `.fai` as an input and only generate it of it's not available? Seems like unnecessary overhead to me

Reply:
Agreed. 
It could be merged with the fasta FYI, `[meta, fasta, fai]`. But kinda of pointless unless you're going to merge all of them on `ch_input`

Reply:
Since I can only implement this or the below suggestion, I went with `ch_genome  // channel: [ val(meta), fasta, fai ]`